src/coord/src/coord/dataflow_builder.rs

```diff
@@ -107,10 +111,10 @@ impl<'a> DataflowBuilder<'a> {
             }
 
             // A valid index is any index on `id` that is known to the dataflow
-            // layer, as indicated by its presence in `self.indexes`.
+            // layer, as indicated by its presence in `self.compute`.
             let valid_index = self.catalog.enabled_indexes()[id]
                 .iter()
-                .find(|(id, _keys)| self.indexes.contains_key(*id));
+                .find(|(id, _keys)| self.compute.collection(*id).is_ok());
```

Comment:
This logic now likely has different behavior than before, as fewer indexes may be known to the controller than were present in `self.indexes`, in hypothetical multi-DDL transactions. Perhaps they never happen, in which case amazing, bullet dodged.